### ISSUE TITLE
NVDA Magnifier: If a fullscreen mode other than "Relative" is used, activating with the mouse some controls in apps fails

### ISSUE THREAD

**Issue by k-kolev1985** [nvaccess/nvda#19674]
### Brief summary

Recently I've noticed that in apps, clicking with the left mouse button (real mouse - not simulation) to activate a control fails for some controls.

Initially, I thought that the problem was in the apps in question. Until last night, when I noticed, that if I don't use the NVDA's built-in magnifier feature, the activation via left mouse click is successful.

Today, I've done some more tests, playing with the settings of the NVDA's magnifier. It turned out, that this is related to the fullscreen mode of the magnifier. If "Center" or "Border" is used, the issue occurs. If the "Relative" mode is used, the issue does not occur.

A good place to test this is the "Discord" app for Windows. On its left side, it has a few buttons to switch between sections such as "Direct Messages", "Add Server", Discover" - clicking with the left mouse buttons should activate each of them, but when the NVDA magnifier is running in center or border fullscreen mode, it does not work. Same applies for some other controls in the main window of "Discord". It applies for some controls in some other apps as well, but I can't at the moment remember in which apps exactly I've noticed it.

@Boumtchack  any ideas what's going on?

### Steps to reproduce

1. Launch NVDA.
2. Install and log in into "Discord".
3. Toggle on the NVDA magnifier with "Center" or "Border" fullscreen mode.
4. Go to the "Discord" app and click with the left mouse button on controls such as "Direct Messages", "Add Server" and "Discover", located on the left edge of the main "Discord" window.
5. Toggle off the NVDA magnifier.
6. Go to the "Discord" app and click with the left mouse button on controls such as "Direct Messages", "Add Server" and "Discover", located on the left edge of the main "Discord" window.
7. Change the default fullscreen mode of the NVDA magnifier to "Relative".
8. Toggle on the NVDA magnifier.
9. Go to the "Discord" app and click with the left mouse button on controls such as "Direct Messages", "Add Server" and "Discover", located on the left edge of the main "Discord" window.

### Actual behavior

In step 4, clicking with the left mouse button fails to activate the control under the mouse cursor.
In step 6, clicking with the left mouse button successfully activates the control under the mouse cursor, as it should.
In step 9, clicking with the left mouse button successfully activates the control under the mouse cursor, as it should.

### Expected behavior

In step 4, clicking with the left mouse button successfully activates the control under the mouse cursor.
In step 6, clicking with the left mouse button successfully activates the control under the mouse cursor, as it should.
In step 9, clicking with the left mouse button successfully activates the control under the mouse cursor, as it should.

### NVDA logs, crash dumps and other attachments

If needs be, I'll provide the log file from NVDA.

### NVDA type

installed copy

### NVDA version

alpha-54718,4bf18781 (2026.2.0.54718)

### Have you tried any other versions of NVDA? If so, please report their behaviors.

No, because the NVDA magnifier feature is not available in previous versions of NVDA.

### Windows version

Windows 11 Pro version 25H2 (build 26200.7840), 64-bit, in Bulgarian with all locale settings set to "Bulgarian".

### Name and version of other software in use when reproducing the issue

Discord version: stable 500334 (b142515) Host 1.0.9225 x64 (75673) Build Override: N/A Windows 11 64-bit (10.0.26200)

### Does the issue still occur after restarting your computer?

I have restarted my computer and the issue still occurs

### If NVDA add-ons are disabled, is your problem still occurring?

I have restarted NVDA with add-ons disabled and the issue still occurs

### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?

I have not run the COM Registration Fixing Tool

**Comment by hwf1324** [comment]
I've encountered similar issues before.

**Comment by seanbudd** [comment]
cc @Boumtchack

**Comment by Boumtchack** [comment]
I'll try some changes, but unfortunalty Discord is not authorized on my machine so I won't be able to test the results.

**Comment by k-kolev1985** [comment]
@Boumtchack 

Is the fix merged into the latest alpha build (alpha-54879,a94c7f85 (2026.2.0.54879)) of NVDA? Because I can still reproduce the issue with it.

Apart from Discord, I've found two more apps with which I can reproduce the issue: HyperX NGENUITY Beta and Microsoft Teams. In Teams, there is a panel on the left side of the main window, which is used to switch between different tabs / sections of the app. Clicking there when the NVDA magnifier is running causes the click not to work.

However, there is an improvement. Now it also works as expected if the fullscreen mode is "Border", but it still does not work with "Center" fullscreen mode.

**Comment by seanbudd** [comment]
Yes it should have been fixed in 54879,a94c7f85. Can you provide a debug log from that version?

**Comment by k-kolev1985** [comment]
OK, here is a debug log. I hope it is the correct one. 😊

[NVDA_Log_2026-02-27_#19674.txt](https://github.com/user-attachments/files/25597608/NVDA_Log_2026-02-27_.19674.txt)

**Comment by hwf1324** [comment]
The "Border" mode is also not working correctly on my end; the category list in the NVDA settings dialog is an excellent test scenario. My resolution is 1920x1080. When clicking on the left half of a category item, the dialog loses focus, although the navigation object updates normally. The dialog is only actually clicked when interacting with the right third of the item.

Strangely, some applications seem to be unaffected, such as Firefox.

**Comment by Boumtchack** [comment]
I changed some logic, it should work properly 
https://github.com/nvaccess/nvda/pull/20052
